src/main/resources/io/viash/platforms/nextflow/workflowFactory/processDirectives.nf

```diff
@@ -0,0 +1,397 @@
+def assertMapKeys(map, expectedKeys, requiredKeys, mapName) {
+  assert map instanceof Map : "Expected argument '$mapName' to be a Map. Found: class ${map.getClass()}"
+  map.forEach { key, val -> 
+    assert key in expectedKeys : "Unexpected key '$key' in ${mapName ? mapName + " " : ""}map"
+  }
+  requiredKeys.forEach { requiredKey -> 
+    assert map.containsKey(requiredKey) : "Missing required key '$key' in ${mapName ? mapName + " " : ""}map"
+  }
+}
+
+// TODO: unit test processDirectives
```

Comment:
What about this TODO?

Reply:
This was already there. atm there is no infrastructure for unit testing individual groovy functions.

I can remove the comment, though.